exonum-java-binding-core/src/main/java/com/exonum/binding/proxy/NativeHandle.java

```diff
@@ -66,11 +69,11 @@ private void checkValid() {
    * Returns true if this native handle is valid.
    */
   final boolean isValid() {
-    return nativeHandle != INVALID_NATIVE_HANDLE;
+    return nativeHandle != INVALID_NATIVE_HANDLE && isValid;
```

Comment:
I wonder if we shall _allow_ accessing `nullptr` == 0, just to simplify the condition here? @vitvakatu , @DarkEld3r , we can't have `nullptr` as a handle anywhere, can we?

Reply:
I believe we can safely use `return isValid` here, yep. We will be able to test debugging capabilities in case of any... hmmm.... troubles :D

Reply:
And what about creating incorrect handles from constructor ? 
new NativeHandel(INVALID_NATIVE_HANDLE) - do we allow this ?

Reply:
I'm not completely sure, but we can allow it

Reply:
@dmitry-timofeev Allow creation of a handle from INVALID_NATIVE_HANDLE ? Should it be VALID until we invalidate it manually ?

Reply:
I'd go with a stricter rule first and relax later if needed: forbid instantiation, if it *ever* (which we currently doubt, if I understand correctly) need that, allow.

Reply:
The idea was to simplify this method — i.e., remove `nativeHandle != INVALID_NATIVE_HANDLE` since this condition is not possible because it is enforced on the instantiation.